src/test/java/org/json/junit/JSONObjectTest.java

```diff
@@ -2025,8 +2026,8 @@ public void valueToString() {
                 "\"key3\":\"val3\""+
              "}";
         JSONObject jsonObject = new JSONObject(jsonObjectStr);
-        assertTrue("jsonObject valueToString() incorrect",
-                JSONObject.valueToString(jsonObject).equals(jsonObject.toString()));
+        JSONAssert.assertEquals("jsonObject valueToString() incorrect",
```

Comment:
Please replace this line with code that converts the strings to JSONObjects and compare them using JSONObject.similar()